### ISSUE TITLE
Operator default of rook/ceph:master is bad idea

### ISSUE THREAD

**Issue by shell-skrimp** [rook/rook#13843]
Similar to #12944 

Helm: v1.13.5
Version: 18.2.1-0

If you `k rollout restart ceph-mon...` the `rook-ceph-operator` never reconciles itself, even if you set `skipUpgradeChecks: true`

This is worrisome because what if these nodes lost power suddenly?

I noticed that mon-a was not visible then I tried to manually start `rook-ceph-mon-a` with `k rollout ...`. I decided to go one step further and restart the rest of the mons and it looks like I'm at the mercy of the operator working correctly?

```
2024-03-01 00:02:56.473146 I | ceph-spec: ceph-file-controller: CephCluster "rook-ceph" found but skipping reconcile since ceph health is &{Health:HEALTH_ERR Details:map[error:{Severity:Urgent Message:failed to get status. . timed out: exit status 1}] LastChecked:2024-03-01T00:01:42Z LastChanged:2024-02-29T23:06:12Z PreviousHealth:HEALTH_WARN Capacity:{TotalBytes:48009345761280 UsedBytes:1680641310720 AvailableBytes:46328704450560 LastUpdated:2024-02-29T23:31:48Z} Versions:<nil> FSID:}
2024-03-01 00:02:56.473192 I | ceph-spec: ceph-fs-subvolumegroup-controller: CephCluster "rook-ceph" found but skipping reconcile since ceph health is &{Health:HEALTH_ERR Details:map[error:{Severity:Urgent Message:failed to get status. . timed out: exit status 1}] LastChecked:2024-03-01T00:01:42Z LastChanged:2024-02-29T23:06:12Z PreviousHealth:HEALTH_WARN Capacity:{TotalBytes:48009345761280 UsedBytes:1680641310720 AvailableBytes:46328704450560 LastUpdated:2024-02-29T23:31:48Z} Versions:<nil> FSID:}
2024-03-01 00:03:06.473699 I | ceph-spec: ceph-file-controller: CephCluster "rook-ceph" found but skipping reconcile since ceph health is &{Health:HEALTH_ERR Details:map[error:{Severity:Urgent Message:failed to get status. . timed out: exit status 1}] LastChecked:2024-03-01T00:01:42Z LastChanged:2024-02-29T23:06:12Z PreviousHealth:HEALTH_WARN Capacity:{TotalBytes:48009345761280 UsedBytes:1680641310720 AvailableBytes:46328704450560 LastUpdated:2024-02-29T23:31:48Z} Versions:<nil> FSID:}
2024-03-01 00:03:06.473746 I | ceph-spec: ceph-fs-subvolumegroup-controller: CephCluster "rook-ceph" found but skipping reconcile since ceph health is &{Health:HEALTH_ERR Details:map[error:{Severity:Urgent Message:failed to get status. . timed out: exit status 1}] LastChecked:2024-03-01T00:01:42Z LastChanged:2024-02-29T23:06:12Z PreviousHealth:HEALTH_WARN Capacity:{TotalBytes:48009345761280 UsedBytes:1680641310720 AvailableBytes:46328704450560 LastUpdated:2024-02-29T23:31:48Z} Versions:<nil> FSID:}
2024-03-01 00:03:12.761208 E | ceph-cluster-controller: failed to get ceph status. failed to get status. . timed out: exit status 1
2024-03-01 00:03:13.300621 I | op-mon: mon a is not yet running
2024-03-01 00:03:13.321545 I | op-mon: mon b is not yet running
2024-03-01 00:03:13.418760 I | op-mon: mon c is not yet running
2024-03-01 00:03:13.418770 I | op-mon: mons running: []
```

**Comment by rkachach** [comment]
> So I believe the issue is that my rook operator was using `rook/ceph:master`. I switched it to `rook/ceph:v1.12.11-18.g582c940e8`.
> 
> I experienced this issue yet again in this helm chart and ceph version.

The latest rook release is `1.13.4` which comes with a lot of fixes. I'd recommend upgrading your setup and see if you are still experimenting issues or not.

**Comment by travisn** [comment]
@shell-skrimp It seems there are two core issues you are asking:

1. You are restarting all the mons at the same time and finding they don't come back up?

The mon quorum is required for the operator to reconcile. Otherwise, the operator is blocked and the ceph cluster is blocked as you found. Why are you restarting the mon pods? The operator is expected to update or restart the mons only when necessary. If all mons do restart at the same time, the quorum is expected to be restored when at least 2/3 of them start up again. If the quorum is not restored (e.g. if majority of mons are lost), then you need to restore the quorum and try the [disaster recovery guide](https://rook.io/docs/rook/latest/Troubleshooting/disaster-recovery/).

2. You are getting the master tag from the helm chart? 

The published helm charts actually use the release tag. The `master` tag is only an artifact of the github repo. Are you installing the published charts according to the [Rook helm docs](https://rook.io/docs/rook/latest/Helm-Charts/helm-charts/)?

**Comment by travisn** [comment]
How exactly did you install the helm chart? Did you follow [these helm instructions](https://rook.github.io/docs/rook/v1.13/Helm-Charts/operator-chart/#release)?
```
helm repo add rook-release https://charts.rook.io/release
helm install --create-namespace --namespace rook-ceph rook-ceph rook-release/rook-ceph -f values.yaml
```

I just installed with these commands and see the latest version (v1.13.5) running. The published charts don't use the master tag so please check the helm install.

**Comment by travisn** [comment]
> Yep I followed it exactly. If you kill some mons/mgr/mds and/or try to update the operator config you should be able to reproduce the issue.

Killing pods wouldn't cause the image to be updated to `master`. How exactly did you "update the operator config"?

> It also looks like the master image has changed since then, so it's possible fixed in some future release. I havent had issues since I locked to a non master release.

The master tag changes every time a PR is merged to master, so it commonly is updated several times a day.

**Comment by travisn** [comment]
Your local rook-ceph-values.yaml must have the `tag:master`. You'll need to remove that so the chart will use the release version. Remove the `tag: master` so that it stops overriding the version. Your values.yaml is only expected to contain the values you want to modify from the released chart.

**Comment by travisn** [comment]
Don't put a version in your values.yaml. The published chart has the correct version. By default you'll get the latest published version (currently v1.13.5). If you want a release version other than the latest, you can use the helm `--version` flag. This would show the released versions: `helm search repo --versions`.

**Comment by travisn** [comment]
Our release process updates the values inside the chart with the release version. I understand the confusion though, so we will look at how to address this.

**Comment by travisn** [comment]
This is fixed now in v1.13.6 with #13897.
